libbeat/monitoring/report/elasticsearch/elasticsearch.go

```diff
@@ -71,8 +71,15 @@ func init() {
 	report.RegisterReporterFactory("elasticsearch", makeReporter)
 }
 
-func makeReporter(beat beat.Info, cfg *common.Config) (report.Reporter, error) {
+func overrideConfigFromBeat(config *config, settings report.Settings) {
+	if settings.DefaultUsername != "" {
+		config.Username = settings.DefaultUsername
+	}
+}
+
+func makeReporter(settings report.Settings, cfg *common.Config) (report.Reporter, error) {
 	config := defaultConfig
+	overrideConfigFromBeat(&config, settings)
```

Comment:
let's replace these with:

```
config := defaultConfig(settings)
```

with:

```
func defaultConfig(settings report.Settings) config {
   ...
}
```

We started with creating default configs as struct and use them as kinda templates by copying the struct. Rethinking this approach I'm in favor of functions producing the default configs.